Cabal/src/Distribution/Types/DumpBuildInfo.hs

```diff
@@ -13,3 +14,14 @@ data DumpBuildInfo
 
 instance Binary DumpBuildInfo
 instance Structured DumpBuildInfo
+
+instance Parsec DumpBuildInfo where
+  parsec = parsecDumpBuildInfo
+
+parsecDumpBuildInfo :: CabalParsing m => m DumpBuildInfo
+parsecDumpBuildInfo = boolToDumpBuildInfo <$> parsec
+
+boolToDumpBuildInfo :: Bool -> DumpBuildInfo
+boolToDumpBuildInfo bool = case bool of
+  True -> DumpBuildInfo
+  _ -> NoDumpBuildInfo
```

Comment:
```suggestion
boolToDumpBuildInfo bool = if bool then DumpBuildInfo else NoDumpBuildInfo
```

```
$ hlint Cabal/src/Distribution/Types/DumpBuildInfo.hs
Cabal/src/Distribution/Types/DumpBuildInfo.hs:(25,28)-(27,22): Suggestion: Use if
Found:
  case bool of
    True -> DumpBuildInfo
    _ -> NoDumpBuildInfo
Perhaps:
  if bool then DumpBuildInfo else NoDumpBuildInfo
```

Reply:
You didn't answer my point. What, in your opinion, is `https://github.com/haskell/cabal/blob/master/.github/workflows/lint.yml`?

Reply:
@geekosaur maybe Phil used a custom config. Config in our repo has a ton of exclusions.

I don't think we should torture Matt sending him a wealth of hlint suggestions this way. We should make sure the PR passes CI with our hlint config, and in a separate PR update config to remove some exclusions and bulk-update the code.

Reply:
That was more or less what I was getting at: there's probably a ton of code elsewhere that has these problems, we should someday update our hlint config and code to match, but it's not worth dying on that hill now.

Reply:
I'll apply the suggestions from your review.